package.json

```diff
@@ -29,7 +30,9 @@
   ],
   "author": "Joel Denning",
   "license": "MIT",
-  "dependencies": {},
+  "dependencies": {
+    "@types/jquery": "latest"
```

Comment:
I think this should be in dev dependencies?

Reply:
Yeah I think so, too. I'm new to typescript, but my understanding is that we don't really need to have people who npm install single-spa to download @types/jquery. The [documentation for DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped#npm) seems to encourage people to use devDependencies instead of dependencies for this.

Reply:
Hi, I believe that the use case here is different. 

We are not simply using types from @types/jquery at development time, but we are packaging a package that exposes external types.

If we move @types/jquery to devDependency we obtain this:
![image](https://user-images.githubusercontent.com/29378614/56483177-19271180-64c0-11e9-8f52-4cd25d2b5440.png)

The reasons are documented here:
- https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html (under Dependencies)
- https://stackoverflow.com/a/46011417

Thoughts?

Reply:
hm, I guess a followup question is -- what value do we get from including the jQuery types? Really all a user of single-spa uses jQuery for is to pass it in and then never uses it again, right? So it almost seems like jQuery's typings aren't even necessary for single-spa?

I could be wrong on this though. Thoughts?

Reply:
Yeah I sorta agree -- almost no one uses the ensureJQuerySupport function, and the most common use case for calling it is actually to just pass in no arguments at all, since most of the time jQuery is on the window anyway and [single-spa can use the window.jQuery variable](https://github.com/CanopyTax/single-spa/blob/master/src/jquery-support.js#L7)

And even when they do it call it with a jQuery object, we don't care what version of jquery it is or what most of the methods are.

I'm leaning towards maybe just making the jQuery argument an optional object? Avoid everyone downloading a types definition that very few would use.

Reply:
You are right, it would work just fine if instead of the jQuery typings, we'd just use `any` or `Object`...
But it would work just fine without typings at all :)

IMHO, since single-spa depends *implicitly* on jQuery given the fact that its API is accepting an optional instance of JQuery, I would make that dependency explicit. Which would be good for static analysis too.

I cannot really see any drawbacks but I am obviously open to whatever you guys consider acceptable.

Reply:
Just so we're all on the same page, the reason that function exists is explained [here](https://single-spa.js.org/docs/api.html#ensurejquerysupport)

Now, personally, I view that function more as a way to get things working for those that do have strange jQuery setups, and 95% (or more?) of the people that use single-spa never call nor even know about this API. 

Which is why I brought up the option of just not adding the typings at all (or Joel's idea of just setting it to `any`).

For someone who is going to call that function, will it provide them any benefit to know the typings of jQuery when all they're really doing is just passing in a reference to it? 

For someone developing single-spa, they should have installed the dev dependencies so that should help with the typings, right? 

Anyway, just my thoughts on this. :)

Reply:
Yeah let's remove it in favor of `any`. The `ensureJQuerySupport` function is used by one codebase that I know of, and that place doesn't even pass in jquery, but lets single-spa find jquery on the window.

I see no situations where this type def helps anyone, and it [causes ~385kb of download](https://unpkg.com/@types/jquery@3.3.29/JQuery.d.ts) for everyone who does `npm install single-spa`. We have a nice clean zero dependencies in our package.json, and this one doesn't seem worth adding to me.

Reply:
Agreed, changed to `any` and dependency removed.
However, 1.3mb sounded too much so I checked and it's 116KB of tarballs
![image](https://user-images.githubusercontent.com/29378614/56555653-78068c80-658d-11e9-9488-40b6fedba110.png)
😄